Add InfluxDB Line Protocol Support, and Add endpoint for JSON

JSMike opened the pull request.

Commit:
Add endpoint for JSON
Update influx module to use line protocol for 0.9
Add warning logging to influx module when receiving a response from the DB
Update syntax for server.coffee, remove parenthesis where not required
Update version to 0.6

Commit:
Add comments

Commit:
Update README.md

Commit:
Add https support

Commit:
Add option to disable http mode
Update documentation

JSMike commented:
Added tag for v0.6.0

zackbloom commented on config/default.yaml:
```diff
@@ -18,7 +27,7 @@ influxdb:
   use_udp: false
   retentionPolicy: 'default'
-  # Acceptable version are: '0.8' and '0.9'
+  # All version other than '0.9' will default to '0.8'
```
This is kind of odd.  It should probably error if you try to enter a value other than '0.8' or '0.9'

JSMike replied:
This is for backwards compatibility. If a user updates their bucky server and copies over their old default.yaml file it will work as it did before the version option was added.

zackbloom commented on config/default.yaml:
```diff
@@ -32,7 +41,7 @@ modules:
   collectors:
   # Uncomment the modules that you'd like to use
-    # - ./modules/collectionLogger
+    - ./modules/collectionLogger
```
Any reason for turning this on by default?

JSMike replied:
In the last PR I sent you I commented them all out, I'd have to double check but I believe I ran into an error if no collectors were available. I also thought it would be best to have the logger for the initial setup so that users can test that the endpoint is being reached before adding database config. I can comment this out again if you'd like.

zackbloom commented on lib/influxdb.coffee:
```diff
@@ -1,78 +1,95 @@
-    database = @config.get('influxdb.database').get() ? 'bucky'
-    username = @config.get('influxdb.username').get() ? 'root'
-    password = @config.get('influxdb.password').get() ? 'root'
+    version = @config.get("influxdb.version").get() ? "0.9"
```
Why double quotes?

JSMike replied:
There was just a lot of mixing between single and double quotes, I wanted to make the quotes more uniform, I just picked one over the other.

JSMike replied:
I've changed these back to single quotes.

zackbloom commented on server.coffee:
```diff
@@ -111,15 +119,29 @@ loadApp = (logger, loadedConfig) ->
+      https = require 'https'
+      fs = require 'fs'
+      httpsOptions = _.mapObject loadedConfig.get('server.https.options').get(), (v, k) ->
+        if _.isString(v)
```
So this is saying any option can also be a file? That seems very dangerous.  What if my string value also happens to be a file? I'm not a big fan of errors which appear and disappear based on what files show up in the local directory.

JSMike replied:
Most https certs are stored in files. The https server requires either the entire cert string or a buffer. With this logic it checks to see if the file exists then we load the contents of the file into a buffer. By default I added an ssl/ path to store the cert files to help prevent the issue you're talking about from happening. I don't believe any of the options that we have to consider would take a forward-slash as part of the option string, so if the file exists then loading its contents should be the correct thing to do.

Here are all of the available options that we'd have to consider:
https://nodejs.org/api/tls.html#tls_tls_createserver_options_secureconnectionlistener

Some additional documentation on express/connect/https
* https://docs.nodejitsu.com/articles/HTTP/servers/how-to-create-a-HTTPS-server
* http://expressjs.com/3x/api.html#app.listen

Note: The only real other option that I see is to replace default.yaml with a custom config module that returns an object. That way we could require 'fs' and allow end-users to add the options similar to the options in the nodejitsu link above and specify when the option should be a file buffer instead of a string. I went with the option that kept default.yaml to make things as backwards compatible as possible.

JSMike replied:
I reviewed this with a co-worker and he agreed with you.

He gave one suggestion for keeping the yaml format, which would be to pass an object containing a specified key, such as filename, with the path.

Example:

```yaml
server:
  port: 5999
  appRoot: "/bucky"
  https:
    port: 5599
    options:
      key:
        filename: "ssl/key.pem"
      cert:
        filename: "ssl/cert.pem"
```

zackbloom commented on README.md:
```diff
@@ -101,7 +101,14 @@ If you need more customization, you can write a module:
 There are a few of types of modules:
 
-
+- Server - Use to set properties of the Bucky Server
```
This is describing a new type of module.  Where is that module defined and used in the source?

JSMike replied:
I'll take another look at this, I believe I was just trying to specify the config options, I may have missed the heading about modules.

zackbloom commented:
Hi @JSMike, thanks for the PR, please see the inline comments.

JSMike commented:
Sure, I'm about to leave the office for the day, I'll look into these tomorrow. Thanks for the feedback.

JSMike commented:
Updated PR caption to DON'T MERGE until the requested changes have been implemented. I'll update this comment and the PR when work is completed.

Commit:
Update configuration section of README.md
Add filePath to https options in default.yaml
Replace double quotes with single quotes in influxdb.coffee
Update logic for loading https config, only load from file when filePath property exists

Commit:
Update download path in readme to match tag
Update package.json to match tag
Add version check for influxdb

JSMike commented:
Updated version to v0.6.2

JSMike commented:
@zackbloom Let me know how this looks. Thank you for your feedback.

edlerd commented on README.md:
```diff
@@ -95,20 +95,82 @@ If you're not already running a stats collection service, you should take a look
+    Use to define the options for https.
+    key and cert are mandatory options, here is a full [list of all available options](https://nodejs.org/api/tls.html#tls_tls_createserver_options_secureconnectionlistener).
+    For all options that accept a buffer you can use the path to the file containing the option's data.
+     - `key:` {Object|String|Buffer}  
```
in the example config below key is a sub-object of server.https.options. Here it is directly under server.https. Which is the right place in the hierarchy for it? Same question goes for cert.

JSMike replied:
@edlerd key and cert are properties of options:
https://nodejs.org/api/https.html#https_https_createserver_options_requestlistener
options is the object passed into https.

sebguilbaud commented:
This PR allows bucky-server to work fine on modern InfluxDB's (1.2.2 at least) as JSON protocol was dropped on 1.0 (https://docs.influxdata.com/influxdb/v0.9/write_protocols/json/)
